src/daemon/rpc.h

```diff
@@ -47,35 +47,41 @@ class t_rpc final
   }
 private:
   cryptonote::core_rpc_server m_server;
+  const std::string m_description;
 public:
   t_rpc(
       boost::program_options::variables_map const & vm
     , t_core & core
     , t_p2p & p2p
+    , const bool restricted
+    , const bool testnet
+    , const std::string port
+    , const std::string description
     )
-    : m_server{core.get(), p2p.get()}
+    : m_server{core.get(), p2p.get()}, m_description{description}
   {
-    MGINFO("Initializing core rpc server...");
-    if (!m_server.init(vm))
+    MGINFO("Initializing " << m_description << " rpc server...");
+
+    if (!m_server.init(vm, restricted, testnet, port))
     {
-      throw std::runtime_error("Failed to initialize core rpc server.");
+      throw std::runtime_error("Failed to initialize " + m_description + " rpc server.");
```

Comment:
I wasn't able to use the `<<` syntax here as it returned an error during compilation related to `invalid operands to binary expression ('const char *' and 'const std::string'...)`

Trying to figure out why (and will replace this once I do figure that out) but just making note of it if anyone knows offhand (still learning C++ so this isn't immediately obvious to me).

Reply:
<< is when the object is something which has this operator, such as a stringtream. operator+ is the right thing to use here.